flash/text/seq2seq/core/data.py

```diff
@@ -0,0 +1,271 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from functools import partial
+from typing import Any, Callable, Optional, Union
+
+from datasets import load_dataset
+from datasets.utils.download_manager import GenerateMode
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from torch import Tensor
+from transformers import AutoTokenizer, default_data_collator
+
+from flash.core.data import DataModule, TaskDataPipeline
+
+
+def prepare_dataset(
```

Comment:
shall this be part (static) [art of the data-pipeline?

Reply:
Eventually it should be! but I've mimicked the text classification pipeline. I think eventually this will be consolidated in time